ultraplot/axes/plot.py

```diff
@@ -2477,54 +2477,43 @@ def _parse_cycle(
-            else:
-                args = (cycle,)
-            cycle = constructor.Cycle(*args, **cycle_kw)
-            with warnings.catch_warnings():  # hide 'elementwise-comparison failed'
-                warnings.simplefilter("ignore", FutureWarning)
-                if return_cycle:
-                    pass
-                elif cycle != self._active_cycle:
-                    self.set_prop_cycle(cycle)
-
-        # Manually extract and apply settings to outgoing keyword arguments
-        # if native matplotlib function does not include desired properties
+        cycle_kw = cycle_kw or {}
         cycle_manually = cycle_manually or {}
-        parser = self._get_lines  # the _process_plot_var_args instance
-        props = {}  # which keys to apply from property cycler
-        # BREAKING in mpl3.9.1 parse has cycle items and no longer posseses _prop_keys
-        for prop, key in cycle_manually.items():
-            if kwargs.get(key, None) is None and any(
-                prop in item for item in parser._cycler_items
-            ):
-                props[prop] = key
-        if props:
-            for dict_ in parser._cycler_items:
-                for prop, key in props.items():
-                    value = dict_[prop]
-                    if (
-                        key == "c"
-                    ):  # special case: scatter() color must be converted to hex
-                        value = pcolors.to_hex(value)
-                    kwargs[key] = value
-
+        cycle_kw.setdefault("N", ncycle)
+
+        # Match-case for cycle resolution
+        match cycle:
+            case None if not cycle_kw:
+                resolved_cycle = None
+            case True:
+                resolved_cycle = constructor.Cycle(rc["axes.prop_cycle"])
+            case str() if cycle.lower() == "none":
+                resolved_cycle = None
+            case str() | int():
+                resolved_cycle = constructor.Cycle(cycle, **cycle_kw)
+            case _:
+                resolved_cycle = None
+
+        # Ignore cycle for single-column plotting
+        resolved_cycle = None if ncycle == 1 else resolved_cycle
+
+        # Return or apply cycle
         if return_cycle:
-            return cycle, kwargs  # needed for stem() to apply in a context()
-        else:
-            return kwargs
+            return resolved_cycle, kwargs
+
+        if resolved_cycle and resolved_cycle != self._active_cycle:
+            self.set_prop_cycle(resolved_cycle)
+
+        # Apply manual cycle properties
+        if cycle_manually:
+            current_prop = self._get_lines._cycler_items[self._get_lines._idx]
+            self._get_lines._idx = (self._get_lines._idx + 1) % len(self._active_cycle)
+            for prop, key in cycle_manually.items():
+                if kwargs.get(key) is None and prop in current_prop:
+                    value = current_prop[prop]
+                    kwargs[key] = pcolors.to_rgba(value) if key == "c" else value
```

Comment:
With the new changes, this block of code is skipped if `return_cycle` is `True`. Why is that?